src/main/java/io/jenkins/plugins/folderauth/misc/PermissionWrapper.java

```diff
@@ -132,4 +132,9 @@ private void checkPermission() {
                 .map(PermissionWrapper::new)
                 .collect(Collectors.toSet());
     }
+
+    @Override
+    public int compareTo(PermissionWrapper other) {
+        return Permission.ID_COMPARATOR.compare(this.permission, other.permission);
```

Comment:
Would it make sense to have a null check just in case?

Reply:
> Would it make sense to have a null check just in case?

@oleg-nenashev From the [JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html#compareTo-T-) for `Comparable`:
> **Throws:**
NullPointerException - if the specified object is null
ClassCastException - if the specified object's type prevents it from being compared to this object.

I've added a `@Nonnull` annotation to the function parameter. Also `other.permission` should not be `null` because the constructor and the `readResolve()` check for null permissions unless someone uses reflection.

Please let me know if it makes sense.